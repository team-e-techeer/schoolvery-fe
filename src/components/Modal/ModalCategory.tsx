import colors from '@/constants/colors';
import { css } from '@emotion/react';
import { useState } from 'react';
import Modal from './Modal';
import Ch from '../../assets/img/ch.png';
import chicken from '../../assets/img/chicken.png';
import Jp from '../../assets/img/jp.png';
import Ko from '../../assets/img/ko.png';
import Pizza from '../../assets/img/pizza.png';
import All from '../../assets/img/all.png';
import Asian from '../../assets/img/asian.png';
import Hamburger from '../../assets/img/hamburger.png';
import koreanFood from '../../assets/img/koreanFood.png';
import coffee from '../../assets/img/coffee.png';
import { useNavigate } from 'react-router-dom';
interface Props {
  visible: boolean;
  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
  isNavigating: boolean;
}

interface Info {
  path: string;
  src: string;
  name: string;
  select: boolean;
  id: number;
}

export default function ModalCategory({ visible, setModalVisible, isNavigating }: Props) {
  const navigate = useNavigate();
  const [imageInfo, setImage] = useState<Info[]>([
    { path: '/category/전체', src: All, name: '전체', select: true, id: 0 },
    { path: '/category/한식', src: Ko, name: '한식', select: false, id: 1 },
    { path: '/category/중식', src: Ch, name: '중식', select: false, id: 2 },
    { path: '/category/일식', src: Jp, name: '일식', select: false, id: 3 },
    { path: '/category/아시안', src: Asian, name: '아시안', select: false, id: 4 },
    { path: '/category/치킨', src: chicken, name: '치킨', select: false, id: 5 },
    { path: '/category/피자', src: Pizza, name: '피자', select: false, id: 6 },
    { path: '/category/햄버거', src: Hamburger, name: '햄버거', select: false, id: 7 },
    { path: '/category/분식', src: koreanFood, name: '분식', select: false, id: 8 },
    { path: '/category/커피', src: coffee, name: '커피', select: false, id: 9 },
  ]);
  const onClickCategory = (image: Info) => {
    setImage(imageInfo.map(img => (img.id === image.id ? { ...img, select: true } : { ...img, select: false })));
    isNavigating && navigate(`${image.path}`);
  };
  return (
    <>
      <Modal visible={visible} setModalVisible={setModalVisible}>
        <>
          <span
            css={css`
              display: flex;
              font-size: 2rem;
              margin: 1rem;
              margin-left: 2%;
            `}
          >
            카테고리를 선택해주세요
          </span>
          <div
            css={css`
              display: flex;
              flex-wrap: wrap;
              box-sizing: border-box;
              justify-content: center;
            `}
          >
            {imageInfo.map(image => (
              <button
                onClick={() => onClickCategory(image)}
                css={css`
                  display: flex;
                  flex-wrap: wrap;
                  border-radius: 0.7rem;
                  box-sizing: border-box;
                  background-color: ${image.select ? colors.mainColor : colors.grey100};
                  padding: 1rem;
                  margin: 2rem 1%;
                  flex: 1 1 30%;
                  justify-content: space-around;
                `}
                key={image.name}
              >
                <img
                  css={css`
                    max-width: 100%;
                    max-height: 3rem;
                  `}
                  src={image.src}
                  alt={`category img for ${image.name}`}
                />
                <span
                  css={css`
                    color: ${image.select ? colors.grey100 : colors.black};
                  `}
                >
                  {image.name}
                </span>
              </button>
            ))}
          </div>
        </>
      </Modal>
    </>
  );
}
