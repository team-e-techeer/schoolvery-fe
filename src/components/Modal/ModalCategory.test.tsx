import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter, Router } from 'react-router-dom';
import ModalCategory from './ModalCategory';
import { createMemoryHistory, MemoryHistory } from 'history';
describe('[Modal Component]', () => {
  it('[ModalCategory Component] 렌더링 테스트', () => {
    const handleClose = jest.fn();
    const { container, getByText } = render(
      <ModalCategory visible={true} setModalVisible={handleClose} isNavigating={false}></ModalCategory>,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(getByText('카테고리를 선택해주세요')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('[ModalCategory Component] 닫기 버튼 누르기', async () => {
    const handleClose = jest.fn();
    const { container } = render(
      <ModalCategory visible={true} setModalVisible={handleClose} isNavigating={false}></ModalCategory>,
      {
        wrapper: BrowserRouter,
      }
    );

    const button = screen.getByTestId('test-modal-button');
    fireEvent.click(button);

    expect(handleClose).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
  it('[ModalCategory Component] 카테고리 이동', async () => {
    const handleClose = jest.fn();
    const history: MemoryHistory = createMemoryHistory();
    const { container } = render(
      <Router location={history.location} navigator={history}>
        <ModalCategory visible={true} setModalVisible={handleClose} isNavigating={true}></ModalCategory>
      </Router>
    );
    history.push('/category/home');

    const chickenIcon = screen.getByText('치킨');
    fireEvent.click(chickenIcon);
    expect(history.location.pathname).toBe('/category/치킨');

    expect(container).toMatchSnapshot();
  });
});
