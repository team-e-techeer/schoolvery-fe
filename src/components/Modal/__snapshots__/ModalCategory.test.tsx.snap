// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Modal Component] [ModalCategory Component] 닫기 버튼 누르기 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2rem;
  margin: 1rem;
  margin-left: 2%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #fc775a;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4 {
  max-width: 100%;
  max-height: 3rem;
}

.emotion-5 {
  color: #f2f4f6;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #f2f4f6;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-8 {
  color: #000000;
}

<div>
  <div
    class="one out"
    id="modal-container"
  >
    <div
      class="modal-background"
    >
      <div
        class="modal"
      >
        <button
          class="emotion-0"
          data-testid="test-modal-button"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
        <span
          class="emotion-1"
        >
          카테고리를 선택해주세요
        </span>
        <div
          class="emotion-2"
        >
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 전체"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              전체
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 한식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              한식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 중식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              중식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 일식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              일식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 아시안"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              아시안
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 치킨"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              치킨
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 피자"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              피자
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 햄버거"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              햄버거
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 분식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              분식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 커피"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              커피
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Modal Component] [ModalCategory Component] 렌더링 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2rem;
  margin: 1rem;
  margin-left: 2%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #fc775a;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4 {
  max-width: 100%;
  max-height: 3rem;
}

.emotion-5 {
  color: #f2f4f6;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #f2f4f6;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-8 {
  color: #000000;
}

<div>
  <div
    class="one"
    id="modal-container"
  >
    <div
      class="modal-background"
    >
      <div
        class="modal"
      >
        <button
          class="emotion-0"
          data-testid="test-modal-button"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
        <span
          class="emotion-1"
        >
          카테고리를 선택해주세요
        </span>
        <div
          class="emotion-2"
        >
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 전체"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              전체
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 한식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              한식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 중식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              중식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 일식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              일식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 아시안"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              아시안
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 치킨"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              치킨
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 피자"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              피자
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 햄버거"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              햄버거
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 분식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              분식
            </span>
          </button>
          <button
            class="emotion-6"
          >
            <img
              alt="category img for 커피"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-8"
            >
              커피
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Modal Component] [ModalCategory Component] 카테고리 이동 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2rem;
  margin: 1rem;
  margin-left: 2%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #f2f4f6;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4 {
  max-width: 100%;
  max-height: 3rem;
}

.emotion-5 {
  color: #000000;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.7rem;
  box-sizing: border-box;
  background-color: #fc775a;
  padding: 1rem;
  margin: 2rem 1%;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-20 {
  color: #f2f4f6;
}

<div>
  <div
    class="one out"
    id="modal-container"
  >
    <div
      class="modal-background"
    >
      <div
        class="modal"
      >
        <button
          class="emotion-0"
          data-testid="test-modal-button"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
        <span
          class="emotion-1"
        >
          카테고리를 선택해주세요
        </span>
        <div
          class="emotion-2"
        >
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 전체"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              전체
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 한식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              한식
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 중식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              중식
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 일식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              일식
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 아시안"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              아시안
            </span>
          </button>
          <button
            class="emotion-18"
          >
            <img
              alt="category img for 치킨"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-20"
            >
              치킨
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 피자"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              피자
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 햄버거"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              햄버거
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 분식"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              분식
            </span>
          </button>
          <button
            class="emotion-3"
          >
            <img
              alt="category img for 커피"
              class="emotion-4"
              src="test-file-stub"
            />
            <span
              class="emotion-5"
            >
              커피
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
